gum/backend-arm64/gumstalker-arm64.c

```diff
@@ -119,12 +738,1491 @@ gum_stalker_add_call_probe (GumStalker * self,
+      immediate_for_sp);
+  STALKER_STORE_REG_INTO_CTX (ARM64_REG_X14, app_stack);
+
+
+  if (type != GUM_PROLOG_MINIMAL)
+  {
+    /* 5) push the instruction pointer */
+    gum_arm64_writer_put_ldr_reg_address (cw, ARM64_REG_X15, GUM_ADDRESS (ip));
+    gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X15, ARM64_REG_X15);
+
+    /* 6) save the stack pointer in the GumCpuContex.sp? */
+    STALKER_STORE_REG_INTO_CTX (ARM64_REG_X15, app_stack);
+    gum_arm64_writer_put_ldr_reg_reg_offset (cw, ARM64_REG_X15, ARM64_REG_SP, G_STRUCT_OFFSET (
+        GumCpuContext,
+        sp));
+  }
+}
+
+static void
+gum_exec_ctx_write_epilog (GumExecCtx * ctx,
+                           GumPrologType type,
+                           GumArm64Writer * cw)
+{
+#if DEBUG_LEVEL == 10
+  g_print ("- gum_exec_ctx_write_epilog - type: %d\n",type);
+#endif
+
+  if (type != GUM_PROLOG_MINIMAL) /* GUM_PROLOG_FULL */
+  {
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X15, ARM64_REG_X15);
+  }
+
+  if (type == GUM_PROLOG_MINIMAL)
+  {
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X30, ARM64_REG_X15);
+    gum_arm64_writer_put_instruction (cw, 0xD51B420F); /* MSR NZCV, X15 */
+
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X29, ARM64_REG_X30);
+
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X16, ARM64_REG_X17);
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X14, ARM64_REG_X15);
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_STALKER_REG_CTX, ARM64_REG_X13);
+
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X4, ARM64_REG_X5);
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X2, ARM64_REG_X3);
+    gum_arm64_writer_put_pop_reg_reg (cw, ARM64_REG_X0, ARM64_REG_X1);
+  }
+  else /* GUM_PROLOG_FULL */
+  {
+    gum_arm64_writer_put_pop_all_q_registers (cw);
+    gum_arm64_writer_put_pop_all_x_registers (cw);
+  }
+
+  /* restore the app_stack (with some tricks) */
+  gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X14, ARM64_REG_X15);
+  gum_arm64_writer_put_mov_reg_reg (cw, ARM64_REG_X14, ARM64_REG_SP);
+
+  gum_arm64_writer_put_ldr_reg_address (cw, ARM64_REG_X15,
+      GUM_ADDRESS (&ctx->app_stack));
+  gum_arm64_writer_put_ldr_reg_reg_offset (cw, ARM64_REG_X15, ARM64_REG_X15, 0);
+  gum_arm64_writer_put_mov_reg_reg (cw, ARM64_REG_SP, ARM64_REG_X15);
+
+  gum_arm64_writer_put_ldr_reg_reg_offset (cw, ARM64_REG_X15, ARM64_REG_X14, 8);
+  gum_arm64_writer_put_ldr_reg_reg_offset (cw, ARM64_REG_X14, ARM64_REG_X14, 0);
+}
+
+static void
+gum_exec_ctx_write_push_branch_target_address (GumExecCtx * ctx,
+                                               const GumBranchTarget * target,
+                                               GumGeneratorContext * gc)
+{
+  GumArm64Writer * cw = gc->code_writer;
+
+#if DEBUG_LEVEL == 10
+  g_print ("gum_exec_ctx_write_push_branch_target_address - enter\n");
+  g_print ("\ttarget is %sdirect\n", (target->is_indirect) ? "in" : "");
```

Comment:
No need for parentheses around a boolean value – it's already clear what the condition is.